### ISSUE TITLE
Keyboard focus when dialog is visible

### ISSUE THREAD

**Issue by AVVI94** [AvaloniaUtils/DialogHost.Avalonia#71]
Hi, could you help investigate a keyboard focus issue? I came across issue #7, which appears to be resolved, but I’ve encountered a problem where keyboard focus is not automatically moved into the popup dialog host when it opens. Instead, focus remains on the previously focused element, allowing users to tab-navigate through the content behind the dialog.

Once the keyboard focus is manually moved into the dialog’s content, it stays confined there as expected. However, if the dialog contains no focusable elements (e.g., in the case of a waiting or progress dialog), focus never moves into the dialog at all.

Could you please look into this behavior?

## Issue:
Keyboard focus does not automatically move to the popup dialog host when it opens, allowing users to tab-navigate through and interact with background elements using the keyboard (e.g., space/enter). Additionally, if the dialog has no focusable elements, focus never moves into the dialog.

## Repro:
1) Checkout the latest commit on the main branch.
2) Run the demo app.
3) Open the dialog using the "PASS MODEL" button and press Shift+Tab.
   * (If you use the "PASS VIEW" button, you can simply press Tab instead.)

## Expected behavior:
The keyboard focus should automatically move into the dialog's content when the popup host or its content is attached to the visual tree.

![image](https://github.com/user-attachments/assets/c98e7cb0-26a2-4179-8e55-151fec443f97)

I was able to partially address the issue with the solution below. However, I’m not sure if this is the correct approach, and it still doesn’t resolve the problem for dialogs without interactive elements.

```cs
    protected virtual void SetKeyboardFocus<T>(T? element) where T : InputElement
    {
        if (element is null)
            return;
        element.AttachedToVisualTree += delegate
        {
            element.Focusable = true;
           // if TabStop is true, the keyboard focus visualisation can be visible, and I dont like it for whole UserControl non-interactive elements...
            element.IsTabStop = false;
            var r = element.Focus();
            Debug.Assert(r);
            // not sure I need this? Is it even correct nav mode?
            KeyboardNavigation.SetTabNavigation(element, KeyboardNavigationMode.Cycle);
        };
    }
```
